projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_connection_base.py

```diff
@@ -96,9 +96,12 @@ def method(*args, **kwargs):
     )
     def connect(self) -> PEP249Connection:
         """
+        Checks if we are still properly connected to the database (including by issuing validation query)
+        and re-connect if not.
```

Comment:
Isn't that performed transparently by the driver? The usual `SELECT 1` mechanism to detect timeouts

Reply:
Depends on the driver library. The point here is the remote end (the database server) may have disconnected or the session may have timed out (on the remote oned) but in the client (here in vdk - we are the client) we do not know. So get_managed_connection as you say aims to return verified connection so "select 1" is statement that both is something that works on each db (we think) and it's usually optimized by most db since it's frequently used by validation